### PR TITLE
Update the minimum external LLVM to 13

### PR DESCRIPTION
**Pull request by cuviper**
With this change, we'll have stable support for LLVM 13 through 15 (pending release).
For reference, the previous increase to LLVM 12 was #90175.

r? @nagisa

### CODE REVIEW

File: compiler/rustc_codegen_llvm/src/builder.rs [hunk 1]
```diff
@@ -1444,51 +1429,37 @@ impl<'a, 'll, 'tcx> Builder<'a, 'll, 'tcx> {
...
+        let name = if let Some(vector_length) = vector_length {
+            format!(
+                "llvm.{}.sat.v{}i{}.v{}f{}",
+                instr, vector_length, int_width, vector_length, float_width
+            )
+        } else {
+            format!("llvm.{}.sat.i{}.f{}", instr, int_width, float_width)
+        };
+        let f = self.declare_cfn(&name, llvm::UnnamedAddr::No, self.type_func(&[src_ty], dest_ty));
+        Some(self.call(self.type_func(&[src_ty], dest_ty), f, &[val], None))
```
Comment (RalfJung) [inline compiler/rustc_codegen_llvm/src/builder.rs:1462]: Does this always return `Some` now?

Comment (cuviper) [reply compiler/rustc_codegen_llvm/src/builder.rs:1462]: For LLVM, yes, and that's passed directly back through `BuilderMethods::fptoui_sat`/`fptosi_sat`. But the GCC version of those methods still returns `None`. So I guess the fallback in cg_ssa will only be used by GCC, and maybe that could be pushed down into something GCC-specific.

Comment (cuviper) [reply compiler/rustc_codegen_llvm/src/builder.rs:1462]: See #100653.

**Pull request merged**